File: tests/agent_workloads/adapters/repository_semantic_edit.mjs
import { cpSync, readFileSync, rmSync, writeFileSync } from "node:fs";

if (process.argv.length !== 3) {
  throw new Error("usage: repository_semantic_edit INPUT_ROOT");
}

const project = "project";
cpSync(process.argv[2], project, { recursive: true });
try {
  const source = `${project}/project/examples/main.jet`;
  const edited = readFileSync(source, "utf8")
    .split(/(?<=\n)/)
    .map((line) => {
      if (line.startsWith("fn prepare()")) return line.replace("fn prepare", "fn configure");
      if (line.replace(/[\r\n]+$/, "") === "    prepare()") {
        return line.replace("prepare()", "configure()");
      }
      return line;
    })
    .join("");
  writeFileSync(source, edited);
  process.stdout.write(readFileSync(source));
} finally {
  rmSync(project, { recursive: true, force: true });
}

File: tests/agent_workloads/adapters/repository_semantic_edit.py
import shutil
import sys
from pathlib import Path


if len(sys.argv) != 2:
    raise SystemExit("usage: repository_semantic_edit INPUT_ROOT")

project = Path("project")
shutil.copytree(sys.argv[1], project)
try:
    source = project / "project" / "examples" / "main.jet"
    lines = source.read_text().splitlines(keepends=True)
    edited = []
    for line in lines:
        if line.startswith("fn prepare()"):
            line = line.replace("fn prepare", "fn configure", 1)
        elif line.rstrip("\r\n") == "    prepare()":
            line = line.replace("prepare()", "configure()", 1)
        edited.append(line)
    source.write_text("".join(edited))
    sys.stdout.write(source.read_text())
finally:
    shutil.rmtree(project)
